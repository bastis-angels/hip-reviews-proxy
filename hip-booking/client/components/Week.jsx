import React from 'react'
import Day from './Day.jsx'

const Row = styled.div`
  display: flex;
  font-size: 1.2em;
  font-weight: 300;
  color: #333333;
  background-color: #fff;
  border: 1px solid #ebebeb;
  border-radius: 0;
  text-align: center;
  text-shadow: none;
`


export default class Week extends React.Component {
  constructor(props) {
    super(props);
    this.getDays.bind(this);
  }

  getDays() {
    let days = [];
    let day = this.props.weekOfDate;
    for (var i = 0; i < 7; i++) {
      days.push(<Day moment={day.clone()} key={i} handleDayClick={this.props.handleDayClick} arrive={this.props.arrive}/>);
      day.add(1, 'day');
    }
    return days;
  }
  render() {
    return (
      <Row>
        {this.getDays()}
      </Row>
    )
  }
}
