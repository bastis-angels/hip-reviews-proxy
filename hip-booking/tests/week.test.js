import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Week from '../client/components/Week.jsx'
import moment from 'moment'


describe('Week', () => {
  it('renders 7 days/child components', () => {
    const week = shallow(
      <Week 
        weekOfDate={moment('2019-03-01').clone().day('Sunday')} 
        arrive={'Select date'} 
        handleDayClick={() => null} 
      />);
    expect(week.getElement().props.children).toHaveLength(7);
  });

  it('renders week based on state.weekOfDate', () => {
    const week = shallow(      
      <Week 
        weekOfDate={moment('2019-05-01').clone().day('Sunday')} 
        arrive={'Select date'} 
        handleDayClick={() => null} 
      />
    );
    expect(toJson(week)).toMatchSnapshot();
  });

});
