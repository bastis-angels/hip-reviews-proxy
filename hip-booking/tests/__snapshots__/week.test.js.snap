// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Week renders week based on state.weekOfDate 1`] = `
<styled.div>
  <Day
    arrive="Select date"
    handleDayClick={[Function]}
    key="0"
    moment={"2019-04-28T07:00:00.000Z"}
  />
  <Day
    arrive="Select date"
    handleDayClick={[Function]}
    key="1"
    moment={"2019-04-29T07:00:00.000Z"}
  />
  <Day
    arrive="Select date"
    handleDayClick={[Function]}
    key="2"
    moment={"2019-04-30T07:00:00.000Z"}
  />
  <Day
    arrive="Select date"
    handleDayClick={[Function]}
    key="3"
    moment={"2019-05-01T07:00:00.000Z"}
  />
  <Day
    arrive="Select date"
    handleDayClick={[Function]}
    key="4"
    moment={"2019-05-02T07:00:00.000Z"}
  />
  <Day
    arrive="Select date"
    handleDayClick={[Function]}
    key="5"
    moment={"2019-05-03T07:00:00.000Z"}
  />
  <Day
    arrive="Select date"
    handleDayClick={[Function]}
    key="6"
    moment={"2019-05-04T07:00:00.000Z"}
  />
</styled.div>
`;
